tailscale/resource_logstream_configuration_test.go

```diff
@@ -0,0 +1,131 @@
+package tailscale_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
+)
+
+const testLogStreamConfiguration = `
+	resource "tailscale_logstream_configuration" "test_logstream_configuration" {
+		log_type         = "configuration"
+		destination_type = "panther"
+		url              = "https://example.com"
+		token            = "some-token"
+	}`
+
+const testLogstreamConfigurationUpdateSameLogtype = `
+	resource "tailscale_logstream_configuration" "test_logstream_configuration" {
+		log_type         = "configuration"
+		destination_type = "cribl"
+		url              = "https://example-other.com"
+		token            = "some-token"
+	}`
+
+const testLogstreamConfigurationUpdateDifferentLogtype = `
+	resource "tailscale_logstream_configuration" "test_logstream_configuration" {
+		log_type         = "network"
+		destination_type = "datadog"
+		url              = "https://example-third.com"
+		token            = "some-token"
+	}`
+
+func TestAccTailscaleLogstreamConfiguration_basic(t *testing.T) {
+	const resourceName = "tailscale_logstream_configuration.test_logstream_configuration"
+
+	checkProperties := func(expectedLogType tsclient.LogType, expectedDestinationType tsclient.LogstreamEndpointType, expectedURL string) func(client *tsclient.Client, rs *terraform.ResourceState) error {
```

Comment:
Nit: instead of passing individual parameters, you could pass a single expected `tsclient.LogstreamConfiguration`.

Reply:
Ahh good call